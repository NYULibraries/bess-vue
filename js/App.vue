<template>
<div class="bobcat_embed">
  <div class="bobcat_embed_tabs_wrapper">
    <div class="bobcat_embed_tabs">
      <ul v-for="tab in tabs" :key="tab.id">
        <li :class="isSelected(tab) ? 'selected' : ''">
          <a :href="tab.href || '#'" :title="tab.title" :alt="tab.alt" :target="tab.target" @click="event => updateTab(event, tab)">{{ tab.label }}</a>
        </li>
      </ul>
    </div>
  </div>
  <div class="bobcat_embed_searchbox">
    <div class="bobcat_embed_tab_content">
      <div class="bobcat_embed_search_field" id="query">
        <search-form :search-key="searchKey" :institution="institution" :vid="vid"></search-form>
      </div>

      <div class="bobcat_embed_links">
        <ul>
          <li v-if="advancedSearch">
            <a target="_blank" :href="advancedSearchLink">Advanced search</a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
</template>

<script>
import qs from 'query-string';
import { primoSearch } from './utils/searchRedirects';
import searchForm from './components/SearchForm.vue';

// source: https://stackoverflow.com/a/4716930/8603212
const queryString = document.currentScript.src.replace(/^[^?]+\??/,'');
const { vid } = qs.parse(queryString);
const { bobcatUrl } = CONFIG;
const { links, advancedSearchLinks, advancedSearch, institution } = CONFIG.institutions[vid];

export default {
  data() {
    return {
      vid,
      institution,
      advancedSearchLink: advancedSearchLinks.books,
      advancedSearch,
      search: '',
      searchKey: 'books',
      tabs: [
        { id: 0, label: 'Books & More', searchKey: 'books' },
        { id: 1, label: 'Articles & Databases', searchKey: 'articles', ...links.articles },
        { id: 2, label: 'Journals', searchKey: 'journals' },
        { id: 3, label: 'Course Reserves', searchKey: 'reserves' },
      ],
    };
  },
  components: {
    searchForm
  },
  methods: {
    openPrimoSearch(search) {
      const url = primoSearch({ search, institution, vid, scope: 'all', tab: 'all', bobcatUrl });
      window.open(url, '_blank');
    },
    updateTab(event, tab) {
      if (!event.srcElement.href) {
        event.preventDefault();
        this.searchKey = tab.searchKey;
      }
    },
    isSelected(tab) {
      return this.searchKey == tab.searchKey;
    }
  },

};
</script>

<style>
.bobcat_embed {

}

.selected {
  font-weight: bold;
}
</style>
