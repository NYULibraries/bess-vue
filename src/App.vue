<template>
  <div class="bobcat_embed">
    <div class="bobcat_embed_tabs_wrapper">
      <div class="bobcat_embed_tabs">
        <ul role="tablist">
          <tab-item
            v-for="tab in tabs"
            :key="tab.id"
            :class="tabClasses( tab )"
            :update-tab="updateTab"
            :tab="tab"
            role="tab"
          >
            {{ tab.label }}
          </tab-item>
        </ul>
      </div>
    </div>
    <div class="bobcat_embed_searchbox">
      <div class="bobcat_embed_tab_content">
        <search-form
          v-if="engine"
          :search-key="selectedTab"
          :engine="engine"
        />
        <div class="bobcat_embed_links">
          <ul>
            <li
              v-for="( html, idx ) in more"
              :key="idx"
              v-html="html"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import SearchForm from './components/SearchForm.vue';
import TabItem from './components/TabItem.vue';

export default {
    components: {
        SearchForm,
        TabItem,
    },
    data() {
        return {
            selectedTab: 1,
            tabs       : this.$config.map( ( tab, idx ) => ( { id: idx + 1, ...tab } ) ),
        };
    },
    computed: {
        tabConfig() {
            return this.tabs.find( tab => tab.id === this.selectedTab );
        },
        more() {
            return this.tabConfig.more;
        },
        engine() {
            return this.tabConfig.engine;
        },
    },
    methods: {
        updateTab( event, tab ) {
            if ( !tab.open ) {
                event.preventDefault();
                this.selectedTab = tab.id;
            }
        },
        tabClasses( tab ) {
            return {
                bobcat_embed_tabs_selected: this.selectedTab === tab.id,
                bobcat_embed_tabs_first   : tab.id === 1,
                bobcat_embed_tabs_inside  : tab.id > 1 && tab.id < this.tabs.length,
                bobcat_embed_tabs_last    : tab.id === this.tabs.length,
            };
        },
    },
};
</script>

