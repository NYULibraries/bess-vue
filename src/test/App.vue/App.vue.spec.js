/* global process */

// `appConfig` and the rendered HTML for App is different when
// `import.meta.env.VITE_DEPLOY_ENV` is not set to "prod", so we have separate
// tests for prod and non-prod deploy environments.
//
// There are various methods for setting the value of `import.meta.env.VITE_DEPLOY_ENV`
// in the config code at runtime.  The methods that involve changing it in test
// code will not serve our purpose because apparently `import appConfig` caches
// the config module, so subsequent attempts to change `import.meta.env.VITE_DEPLOY_ENV`
// in the configuration after initial import all fail.  We work around this
// problem by doing separate runs of this test file for VITE_DEPLOY_ENV=prod and
// VITE_DEPLOY_PROD_ENV left undefined, which is the value of the environment
// variable when in test mode.
//
// Note that a different set of snapshots is produced/checked for each
// VITE_DEPLOY_ENV, so for each test run there will be spurious warnings of
// obsolete snapshots (i.e. those for VITE_DEPLOY_ENV values other than the
// one being tested).
// Example -- for the VITE_DEPLOY_ENV=prod test run, there will be warnings
// for the snapshots for VITE_DEPLOY_ENV=undefined, which will appear to
// vitest to be obsolete:
//
// Snapshots  16 obsolete
//              ↳ src/test/App.vue/App.vue.spec.js
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU' > engine search > should call `window.open` with correct URL > for all-whitespace search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU' > engine search > should call `window.open` with correct URL > for empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU' > engine search > should call `window.open` with correct URL > for non-empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU' > has the correct HTML 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU_HOME' > engine search > should call `window.open` with correct URL > for all-whitespace search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU_HOME' > engine search > should call `window.open` with correct URL > for empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU_HOME' > engine search > should call `window.open` with correct URL > for non-empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYU_HOME' > has the correct HTML 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUAD' > engine search > should call `window.open` with correct URL > for all-whitespace search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUAD' > engine search > should call `window.open` with correct URL > for empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUAD' > engine search > should call `window.open` with correct URL > for non-empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUAD' > has the correct HTML 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUSH' > engine search > should call `window.open` with correct URL > for all-whitespace search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUSH' > engine search > should call `window.open` with correct URL > for empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUSH' > engine search > should call `window.open` with correct URL > for non-empty search 1
//                · App [ VITE_DEPLOY_ENV: undefined ] > 'NYUSH' > has the correct HTML 1
//
import { beforeEach, describe, expect, test, vi } from 'vitest';
import { config, mount } from '@vue/test-utils';
import App from '../../App.vue';
import appConfig from '../../../config/';

describe( `App [ VITE_DEPLOY_ENV: ${ process.env.VITE_DEPLOY_ENV } ]`, () => {
    let wrapper;

    describe.each(
        Object.keys( appConfig.institutions ).map(
            function ( institution ) {
                return { institution };
            },
        ) )( '$institution', ( { institution } ) => {
        beforeEach( () => {
            config.global.mocks = {
                $config: appConfig.institutions[ institution ],
            };

            wrapper = mount( App );
        } );

        test( 'has the correct HTML', () => {
            expect( wrapper.html() ).toMatchSnapshot();
        } );

        // This test suite assumes that the default tab is the engine search tab.
        describe( 'engine search', () => {
            // A single TAB followed by three spaces
            const ALL_WHITESPACE_SEARCH = '    ';
            const EMPTY_SEARCH = '';
            const NON_EMPTY_SEARCH = 'art';

            const testState = {
                target: undefined,
                reset() {
                    this.target = undefined;
                },
            };

            // This is used for the `describe.runIf` test conditions below.
            // Originally tried to do something like this:
            //
            //    describe.runIf( appConfig.institutions[ institution ][ 0 ].ui?.searchScopeDropdown )( ...
            //        describe.each( [...appConfig.institutions[ institution ][ 0 ].ui.searchScopeDropdown.options.map( option => { searchScopeDropdownValue: option } )( ...
            //
            // ...but it looks like the `describe.each` test is evaluated whether
            // the `describe.runIf` condition is truthy or not, leading to errors
            // when trying to read the `searchScopeDropdown` property of an
            // undefined `appConfig.institutions[ institution ][ 0 ].ui`.
            let currentConfig = appConfig.institutions[ institution ];
            let searchScopeDropdownOptionValues = currentConfig[ 0 ].ui?.searchScopeDropdown.options ?
                Object.keys( currentConfig[ 0 ].ui?.searchScopeDropdown.options ) :
                [];

            beforeEach( () => {
                // Suppress error "Error: Not implemented: window.open", which doesn't
                // cause the test to fail but does make test output harder to read.
                vi.stubGlobal( 'open', vi.fn( ( location ) => {
                    testState.called = true;
                    testState.target = location;
                } ) );

                config.global.mocks = {
                    $config: appConfig.institutions[ institution ],
                };

                wrapper = mount( App );

                testState.reset();
            } );

            describe.runIf( searchScopeDropdownOptionValues.length === 0 )( 'should call `window.open` with correct URL', () => {
                test.each(
                    [
                        { title: 'empty search', inputValue: EMPTY_SEARCH },
                        { title: 'all-whitespace search', inputValue: ALL_WHITESPACE_SEARCH },
                        { title: 'non-empty search', inputValue: NON_EMPTY_SEARCH },
                    ] )( '$title: $inputValue', async ( { inputValue } ) => {
                    expect( testState.target ).toBeUndefined();

                    const form = wrapper.find( 'form' )
                    const input = form.find( 'input' );
                    await input.setValue( inputValue );
                    form.trigger( 'submit' );

                    expect( testState.target ).toMatchSnapshot();
                } )
            } );

            describe.runIf( searchScopeDropdownOptionValues.length > 0 )( 'should call `window.open` with correct URL', () => {
                describe.each(
                    [
                        ...searchScopeDropdownOptionValues.map( function ( option ) {
                            return { 'searchScopeDropdownValue': option }
                        } ),
                    ],
                )(
                    'for search scope dropdown value: $searchScopeDropdownValue', ( { searchScopeDropdownValue } ) => {
                        test.each(
                            [
                                { title: 'empty search', inputValue: EMPTY_SEARCH },
                                { title: 'all-whitespace search', inputValue: ALL_WHITESPACE_SEARCH },
                                { title: 'non-empty search', inputValue: NON_EMPTY_SEARCH },
                            ] )( '$title: $inputValue', async ( { inputValue } ) => {
                            expect( testState.target ).toBeUndefined();

                            const form = wrapper.find( 'form' )
                            const searchScopeDropdown = form.find( 'select' );
                            const input = form.find( 'input' );
                            await searchScopeDropdown.setValue( searchScopeDropdownValue );
                            await input.setValue( inputValue );
                            form.trigger( 'submit' );

                            expect( testState.target ).toMatchSnapshot();
                        } )
                    },
                );
            } );
        } );
    } );
} );
