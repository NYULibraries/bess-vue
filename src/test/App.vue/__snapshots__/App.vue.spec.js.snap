// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App [ VITE_DEPLOY_ENV: undefined ] > 'NYU' > has the correct HTML 1`] = `
"<div class="bobcat_embed">
  <div class="bobcat_embed_tabs_wrapper">
    <div class="bobcat_embed_tabs">
      <ul role="tablist">
        <li class="bobcat_embed_tabs_selected bobcat_embed_tabs_first" role="tab"><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings">Catalog Search</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank">Databases</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="https://guides.nyu.edu" title="Guides to help you find library resources on specific subjects and courses" target="_blank">Research Guides</a></li>
        <li class="bobcat_embed_tabs_last" role="tab"><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank">Course Reserves</a></li>
      </ul>
    </div>
  </div>
  <div class="bobcat_embed_searchbox">
    <div class="bobcat_embed_tab_content">
      <form class="tab-1-search-form">
        <div class="bobcat_embed_search_field"><span class="bobcat_embed_"><label for="tab-1-query">Search for </label><input id="tab-1-query" aria-label="Search Bobcat" type="text" class="bobcat_embed_searchbox_textfield" placeholder="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR" aria-describedby="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR"></span><!-- Dropdown for selecting search scope --><select id="tab-1-scope" class="bobcat_embed_select_value" aria-label="Select search scope">
            <option value="CI_NYU_CONSORTIA"> Library catalog </option>
            <option value="NYU_CONSORTIA"> Library catalog (excluding articles) </option>
            <option value="ARTICLES"> Articles </option>
            <option value="NYUBAFC"> NYU Avery Fisher Center (A/V materials) </option>
            <option value="NYUSC"> NYU Special Collections </option>
          </select><span class="bobat_embed_searchbox_submit_container"><input aria-label="Search" class="bobcat_embed_searchbox_submit" name="Submit" type="submit" value="GO"></span></div>
      </form>
      <div class="bobcat_embed_links">
        <ul>
          <li><a href="https://search.library.nyu.edu/discovery/search?vid=01NYU_INST:NYU_DEV&amp;lang=en&amp;mode=advanced" target="_blank">Advanced search</a></li>
          <li>Need the full text of an article? <a href="https://search.library.nyu.edu/discovery/citationlinker?vid=01NYU_INST:NYU_DEV" target="_blank">Use the search by citation tool</a>.</li>
        </ul>
      </div>
    </div>
  </div>
</div>"
`;

exports[`App [ VITE_DEPLOY_ENV: undefined ] > 'NYU_HOME' > has the correct HTML 1`] = `
"<div class="bobcat_embed">
  <div class="bobcat_embed_tabs_wrapper">
    <div class="bobcat_embed_tabs">
      <ul role="tablist">
        <li class="bobcat_embed_tabs_selected bobcat_embed_tabs_first" role="tab"><a href="https://search.library.nyu.edu/discovery/search?vid=01NYU_INST:NYU_DEV" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings" target="_blank">Catalog Search</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank">Articles &amp; Databases</a></li>
        <li class="bobcat_embed_tabs_last" role="tab"><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank">Course Reserves</a></li>
      </ul>
    </div>
  </div>
  <div class="bobcat_embed_searchbox">
    <div class="bobcat_embed_tab_content">
      <form class="tab-1-search-form">
        <div class="bobcat_embed_search_field"><span class="bobcat_embed_"><label for="tab-1-query">Search for </label><input id="tab-1-query" aria-label="Search Bobcat" type="text" class="bobcat_embed_searchbox_textfield" placeholder="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR" aria-describedby="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR"></span><!-- Dropdown for selecting search scope --><select id="tab-1-scope" class="bobcat_embed_select_value" aria-label="Select search scope">
            <option value="CI_NYU_CONSORTIA"> Library catalog </option>
            <option value="NYU_CONSORTIA"> Library catalog (excluding articles) </option>
            <option value="ARTICLES"> Articles </option>
            <option value="NYUBAFC"> NYU Avery Fisher Center (A/V materials) </option>
            <option value="NYUSC"> NYU Special Collections </option>
          </select><span class="bobat_embed_searchbox_submit_container"><input aria-label="Search" class="bobcat_embed_searchbox_submit" name="Submit" type="submit" value="GO"></span></div>
      </form>
      <div class="bobcat_embed_links">
        <ul>
          <li><a href="https://search.library.nyu.edu/discovery/search?vid=01NYU_INST:NYU_DEV&amp;mode=advanced" target="_blank">Advanced search</a></li>
          <li><a href="https://search.library.nyu.edu/discovery/citationlinker?vid=01NYU_INST:NYU_DEV" target="_blank">For full text articles use the search by citation tool</a></li>
          <li><a href="https://search.library.nyu.edu/discovery/account?vid=01NYU_INST:NYU_DEV&amp;section=overview" target="_blank" class="external-link">My Library Account</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>"
`;

exports[`App [ VITE_DEPLOY_ENV: undefined ] > 'NYUAD' > has the correct HTML 1`] = `
"<div class="bobcat_embed">
  <div class="bobcat_embed_tabs_wrapper">
    <div class="bobcat_embed_tabs">
      <ul role="tablist">
        <li class="bobcat_embed_tabs_selected bobcat_embed_tabs_first" role="tab"><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings">Catalog Search</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="http://guides.nyu.edu/adarch" title="Search databases for articles or browse databases by subject" target="_blank">Articles &amp; Databases</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="#" title="Search for library materials that are held at one location for a particular course">Course Reserves</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="#" title="Guides to help you find library resources on specific subjects and courses">Research Guides</a></li>
        <li class="bobcat_embed_tabs_last" role="tab"><a href="#" title="My Accounts">My Accounts</a></li>
      </ul>
    </div>
  </div>
  <div class="bobcat_embed_searchbox">
    <div class="bobcat_embed_tab_content">
      <form class="tab-1-search-form">
        <div class="bobcat_embed_search_field"><span class="bobcat_embed_"><label for="tab-1-query">Search for </label><input id="tab-1-query" aria-label="Search Bobcat" type="text" class="bobcat_embed_searchbox_textfield" placeholder="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR" aria-describedby="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR"></span><!-- Dropdown for selecting search scope --><select id="tab-1-scope" class="bobcat_embed_select_value" aria-label="Select search scope">
            <option value="CI_NYU_CONSORTIA"> Library catalog </option>
            <option value="NYU_CONSORTIA"> Library catalog (excluding articles) </option>
            <option value="ARTICLES"> Articles </option>
            <option value="NYUBAFC"> NYU Avery Fisher Center (A/V materials) </option>
            <option value="NYUSC"> NYU Special Collections </option>
          </select><span class="bobat_embed_searchbox_submit_container"><input aria-label="Search" class="bobcat_embed_searchbox_submit" name="Submit" type="submit" value="GO"></span></div>
      </form>
      <div class="bobcat_embed_links">
        <ul>
          <li><a href="https://search.abudhabi.library.nyu.edu/discovery/search?vid=01NYU_AD:AD_DEV&amp;mode=advanced" target="_blank">Advanced search</a></li>
          <li>Need the full text of an article? <a href="https://search.abudhabi.library.nyu.edu/discovery/citationlinker?vid=01NYU_AD:AD_DEV" target="_blank">Use the search by citation tool</a>.</li>
        </ul>
      </div>
    </div>
  </div>
</div>"
`;

exports[`App [ VITE_DEPLOY_ENV: undefined ] > 'NYUSH' > has the correct HTML 1`] = `
"<div class="bobcat_embed">
  <div class="bobcat_embed_tabs_wrapper">
    <div class="bobcat_embed_tabs">
      <ul role="tablist">
        <li class="bobcat_embed_tabs_selected bobcat_embed_tabs_first" role="tab"><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings">Catalog Search</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank">Articles &amp; Databases</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank">Course Reserves</a></li>
        <li class="bobcat_embed_tabs_inside" role="tab"><a href="#" title="Guides to help you find library resources on specific subjects and courses">Research Guides</a></li>
        <li class="bobcat_embed_tabs_last" role="tab"><a href="#" title="My Accounts">My Accounts</a></li>
      </ul>
    </div>
  </div>
  <div class="bobcat_embed_searchbox">
    <div class="bobcat_embed_tab_content">
      <form class="tab-1-search-form">
        <div class="bobcat_embed_search_field"><span class="bobcat_embed_"><label for="tab-1-query">Search for </label><input id="tab-1-query" aria-label="Search Bobcat" type="text" class="bobcat_embed_searchbox_textfield" placeholder="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR" aria-describedby="&quot;disability in higher education&quot;, Journal of Medicine, JSTOR"></span><!-- Dropdown for selecting search scope --><select id="tab-1-scope" class="bobcat_embed_select_value" aria-label="Select search scope">
            <option value="CI_NYU_CONSORTIA"> Library catalog </option>
            <option value="NYU_CONSORTIA"> Library catalog (excluding articles) </option>
            <option value="ARTICLES"> Articles </option>
            <option value="NYUBAFC"> NYU Avery Fisher Center (A/V materials) </option>
            <option value="NYUSC"> NYU Special Collections </option>
          </select><span class="bobat_embed_searchbox_submit_container"><input aria-label="Search" class="bobcat_embed_searchbox_submit" name="Submit" type="submit" value="GO"></span></div>
      </form>
      <div class="bobcat_embed_links">
        <ul>
          <li><a href="https://search.shanghai.library.nyu.edu/discovery/search?vid=01NYU_US:SH_DEV&amp;mode=advanced" target="_blank">Advanced search</a></li>
          <li>Need the full text of an article? <a href="https://search.shanghai.library.nyu.edu/discovery/citationlinker?vid=01NYU_US:SH_DEV" target="_blank">Use the search by citation tool</a>.</li>
        </ul>
      </div>
    </div>
  </div>
</div>"
`;
